client/my-sites/media-library/drop-zone.jsx

```diff
@@ -35,9 +39,13 @@ module.exports = React.createClass( {
 			return;
 		}
 
-		MediaActions.clearValidationErrors( this.props.site.ID );
-		MediaActions.add( this.props.site.ID, files );
-		this.props.onAddMedia();
+		this.props.onBeforeAddMedia( files )
+			.then(() => {
```

Comment:
Also, needs spaces around function arguments ([reference](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#arrays-and-function-calls)).